### ISSUE TITLE
Custom fonts don't work

### ISSUE THREAD

**Issue by scott-cornwell** [pixijs/pixijs#9412]
### Current Behavior

Loading fonts with Assets does not seem to work on Pixi 7.2.4:

See Pixi Playground example:
https://www.pixiplayground.com/#/edit/N1-USCnllrXJHgKHh3T6d

I originally tried this locally and it does not work there either.

### Expected Behavior

The correct font should render, it just renders what I think is the default serif font even though Roboto is set and the font is loaded before the text is added. I even set the @font-face in CSS although I'm not sure if that is actually needed (although according to this example it is not mentioned anywhere https://pixijs.io/examples/#/text/from-font.js).

### Steps to Reproduce

https://www.pixiplayground.com/#/edit/N1-USCnllrXJHgKHh3T6d

### Environment

- **`pixi.js` version**: 7.2.4
- **Browser & Version**: Chrome 112.0.5615.137
- **OS & Version**: Windows 10
- **Running Example**:  https://www.pixiplayground.com/#/edit/N1-USCnllrXJHgKHh3T6d

**Comment by scott-cornwell** [comment]
Correction: They do work, but at a minimum the Example here could use updating: https://pixijs.io/examples/#/text/from-font.js

I've updated the example to use the hash in the filename from google cdn so it actually works and *is* Roboto now.

![image](https://user-images.githubusercontent.com/4692594/235321861-898ca47c-8cfa-41a6-8358-ab118e0656f4.png)

It seems that the name you must use in TextStyle for fontFamily is the *filename*. There seems to be filtering involved because using this hash directly from the screenshot above works, but locally if my filename is `Roboto-Regular.ttf` the `family` gets set as just `Roboto Regular`. The example would lead one to believe it goes by the name you set in the bundle.

**Comment by SuperSodaSea** [comment]
Yes, we do need to write better documentation for this. You can find the default font family naming logic here:

<https://github.com/pixijs/pixijs/blob/5f6b33536dfd4a610e97d4bf94c2b86663e36eb7/packages/assets/src/loader/parsers/loadWebFont.ts#L47-L79>

You can get default font family name like this (The result of `await Assets.load('Roboto')` is a [`FontFace`](https://developer.mozilla.org/en-US/docs/Web/API/FontFace)):

```js
const fontFamily = (await Assets.load('Roboto')).family;
```

Or you can specify the font family name explicitly like this:

```js
Assets.addBundle('fonts', {
    Roboto: {
        src: 'https://fonts.gstatic.com/s/roboto/v30/KFOmCnqEu92Fr1Mu4mxK.woff2',
        data: { family: 'Roboto Regular' },
    },
});
```

**Comment by kreso22** [comment]
> ```js
> Assets.addBundle('fonts', {
>     Roboto: {
>         src: 'https://fonts.gstatic.com/s/roboto/v30/KFOmCnqEu92Fr1Mu4mxK.woff2',
>         data: { family: 'Roboto Regular' },
>     },
> });
> ```

Couldn't get Text to work, spend few hours using different fonts, families, formats, web hosts...
This info was crucial for me.
